File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        if(k==0) return false;
        Map<Integer, Integer> mp = new HashMap<>();
        for(int i =0;i<nums.length;i++){
            int j = nums[i];
            if(mp.containsKey(j) && i - mp.get(j)<= k) return true;
            mp.put(j,i);
        }
        return false;  
    }
}

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function(nums, k) {
    const mp = new Map();
    for(let i =0;i<nums.length;i++){
        if(i-mp.get(nums[i])<=k) return true;
        mp.set(nums[i],i);
    }
    return false;
};